src/useSubscription/useSubscription.spec.ts

```diff
@@ -555,4 +555,40 @@ describe('subscribe', () => {
     expect(action).toBeCalledTimes(2)
   })
 
+  it('unsubscribes nested subscriptions automatically', () => {
+    const manager = new Manager()
+    const action = vi.fn()
+    let childSubscription: UseSubscription<typeof action>
+
+    const subscription = useSubscription((): void => {
+      childSubscription = useSubscription(action)
+    }, [], { manager })
+
+    const spy = vi.spyOn(childSubscription!, 'unsubscribe')
+
+    subscription.unsubscribe()
+
+    expect(spy).toBeCalledTimes(1)
+  })
+
+  it('changing args unsubscribes nested subscriptions automatically', async () => {
+    const number = ref(0)
+    const manager = new Manager()
+    const action = vi.fn()
+    let childSubscription: UseSubscription<typeof action>
+
+    useSubscription((number: number): number => {
+      childSubscription = useSubscription(action)
+      return number
+    }, [number], { manager })
+
+    const spy = vi.spyOn(childSubscription!, 'unsubscribe')
+
+    number.value++
+
+    await timeout()
+
+    expect(spy).toBeCalledTimes(1)
```

Comment:
I think it's obvious, but just to confirm we expect child to be unsubscribed _and replaced with a new subscription_?

Reply:
I don't think it will be replaced? Its just unsubscribed. The only way it would get replaced is if the args to that subscription changed.